### ISSUE TITLE
"Run all" command does not work anymore

### ISSUE THREAD

**Issue by castellco** [rstudio/rstudio#11313]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2022.02.2-485
    OS Version      :  Windows 11
    R Version       : 4.2.0

### Steps to reproduce the problem
- Open a RMarkdown document with some chunks. Mine has ~30 chunks.
- Click the Run all, Run chunks below, Restart R and run all chunks, etc. buttons. 
### Describe the problem in detail
Nothing happens when clicking those buttons. The only ones that do work are the "Run current chunk" anad "Run next chunk".
### Describe the behavior you expected
They should work as expected.

- [ x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [ x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
[diagnostics-report.txt](https://github.com/rstudio/rstudio/files/8788818/diagnostics-report.txt)

**Comment by kevinushey** [comment]
Thanks for the bug report! I wasn't able to reproduce using the latest [daily builds](https://dailies.rstudio.com), with the following document:


`````
---
title: "Chunks"
output: html_document
---

```{r}
1
```

```{r}
2
```

```{r}
3
```
`````

For posterity:

```
    RStudio Edition : 2022.6.0.421
    RStudio Version : Desktop [Open Source]
    OS Version      : macOS Monterey 12.4
    R Version       : R version 4.2.0 (2022-04-22)
```

Are you able to provide any other detail that might help us reproduce the problem locally? (Perhaps the issue is somehow specific to the content in your R Markdown document?)

**Comment by Robinlovelace** [comment]
On the latest version if there is a block like the following it fails for me:

```{block2 02-spatial-data-5, type="rmdnote"}
Take care when using the word 'vector' as it can have two meanings in this book:
geographic vector data and the `vector` class (note the `monospace` font) in R.
The former is a data model, the latter is an R class just like `data.frame` and `matrix`.
Still, there is a link between the two: the spatial coordinates which are at the heart of the geographic vector data model can be represented in R using `vector` objects.
```

**Comment by ronblum** [comment]
@gisselacornejocastellano and @Robinlovelace A couple of questions:
1. Did this work in previous versions of R and/or RStudio, and if so, which ones?
2. Is this only an issue when using `block2` or also with `r` or other chunk types?
3. I hate to ask, but I've never used `block2` before—is there any setup I need to perform, or is this part of a project type such as `bookdown` etc., to get it to work? I can see that the running the chunk is attempted when run on its own or with any of the "run all" options, but I get:
    `Warning: don't know how to handle 'block2' engine output`

**Comment by ronblum** [comment]
The tag "repro needed" really means, "more information needed," or more fully, "internal RStudio employees can't reproduce it, so a reprex or other information is needed from the original poster or anyone in the user community." But that would be a very long tag. 😄

**Comment by ronblum** [comment]
I don't run into the issue with the Rmd, either. I'll try building the book and see what happens, just to see if I can get `block2` to work at all. I have to install a bunch of packages in different pages, though, which inspired me to file enhancement request #11321.

**Comment by Robinlovelace** [comment]
Interesting... Will re-test on latest version of RStudio here and share the vid in case that would be useful.

**Comment by Robinlovelace** [comment]
Illustration of the issue I'm hitting, may be different from the one originally described but confident that it's reproducible, unless it's been fixed in more recent versions:

![Peek 2022-05-31 17-21](https://user-images.githubusercontent.com/1825120/171224156-644a5529-a35a-45d2-8f54-1cc69b873f50.gif)

**Comment by gisselacornejocastellano** [comment]
Hi, sorry for my late response. I tried to run the same Rmd again and the problem persists. To explain a little bit more, I hit "Run All" and nothing happens in a Rmd that starts like this:

---
title: "Homework"
author: "Me"
date: '2022-05-24'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
## First subtitle [...]


However, I sent this same Rmd to a classmate, who edited it and resent it to me. That document starts like this:

---
title: "Homework"
author: "we both"
date: "24/05/2022"
output: 
  rmdformats::material:
  highlight: kate
  self_contained: true
  code_folding: show
  thumbnails: true
  gallery: true
  fig_width: 4
  fig_height: 4
  df_print: kable
  pkgdown:
  as_is: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

**Authors** 
[...]

And it DOES work when hitting "Run all". All the chunks are executed correctly. I haven't tried to execute the same Rmd in other versions or Rstudio.

**Comment by Robinlovelace** [comment]
Update on this: it is for sure a regression. I recently updated RStudio from the latest stable version to the latest daily to test #11198. I tested if this had been fixed with the following reprex:

Open an Rmd file. Add the following chunk (see original here):

```
{block2 01-introduction-1, type='rmdnote'}
Reproducibility is a major advantage of command-line interfaces, but what does it mean in practice?
We define it as follows: "A process in which the same results can be generated by others using publicly accessible code."
```

And then

```
1 + 1
```

(With the 3 back ticks of course).

Then try running all by clicking this button:
![image](https://user-images.githubusercontent.com/1825120/173960038-316afdd5-e938-40bb-9c93-c3e03bb806cb.png)

That results in the following:

```
> # Chunk 1: 01-introduction-1
> Reproducibility is a major advantage of command-line interfaces, but what does it mean in practice?
Error: unexpected symbol in "Reproducibility is"
```

It is clear that the bug is caused by R trying and failing to run the block as R code when in fact it is not because it does not contain `{r}` after the tripple back ticks. This is a regression. Please try to reproduce and tag it as such if you can reproduce this bug @ronblum and team. Many thanks!

**Comment by ronblum** [comment]
@Robinlovelace Is this an issue with python, bash, or other non-R?

@gisselacornejocastellano For the example you provide up above at https://github.com/rstudio/rstudio/issues/11313#issuecomment-1142678579, can you provide an example of a full Rmd that doesn't run? Just as minimal example would work well—or even if it turns out that the default R Markdown file doesn't work.

@jgutman Do you have any suggestions for reproducing the issue?

**Comment by ronblum** [comment]
Also, for running other types, I mean even something simple like this with bash, since `pwd` would fail if run as R.

````
---
title: "Trying out bash"
author: "Ron Blum"
date: '2022-06-16'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

First bash
```{bash}
pwd
```

Then R
```{r}
x <- 1
x
```
````

**Comment by ronblum** [comment]
@Robinlovelace Do you have "Chunk Output in Console" set by chance? I can reproduce the problem in that case, and it works with "Chunk Output Inline".

@jgutman reminded me about this behavior, tracked in https://github.com/rstudio/rstudio/issues/10677.

**Comment by Robinlovelace** [comment]
> Do you have "Chunk Output in Console" set by chance?

Correct :+1:

**Comment by Robinlovelace** [comment]
Inline should be the default according to evidence presented here (scroll down from a poll of users): https://github.com/rstudio/rstudio/issues/5280

**Comment by jgutman** [comment]
Yes, unfortunately having that setting checked breaks a whole slew of behaviors in rmarkdown, that only work when chunk output is shown inline. We hope to change that behavior soon, and have several issues tracking that, but in the meantime, unfortunately the only workaround is to set your chunk output to inline. Otherwise in console mode the chunks currently only run one at a time.

**Comment by Robinlovelace** [comment]
Could a quick fix be simply to ignore and not run non `r` chunks when not in the inline mode?

**Comment by jgutman** [comment]
Yeah, that's a reasonable workaround. Due to where we are now in the release cycle (a new release of RStudio will be coming shortly), we can't make any commitments right now as to when we will implement even a quick fix. You're right that that is the source of the issue, as the console can only execute R code, that it is the combination of being in console mode and executing non-R chunks that causes the problem with Run all. Thanks for raising this issue, we'll triage this for consideration of when we can implement the fix!

**Comment by kevinushey** [comment]
Just to double check, is this a regression? (Does this work as expected in the current release?)

If so, I think it's worth fixing this for the next release, assuming we can have high confidence in the fix. I think we have a fairly good understanding of what fix needs to be made here?

If not (or if we want to remain cautious) we could instead target this for a SW patch release.

**Comment by jgutman** [comment]
It's not a regression. Running a Rmarkdown in console mode has never had the ability to handle non-R chunks, because it sends them all to the console where they generally fail due to not being R code. I just tested this again in `2022.02.4+495` to confirm.

Set "Chunk output in Console" and hit Run all
~~~~
```{r cars}
summary(cars)
```

```{python}
name = "Jackie"
print(f"hello {name}")
```

```{bash}
pwd
```


~~~~

**Comment by kevinushey** [comment]
Thanks for confirming! In that case, I agree we should leave it out of SW.

**Comment by Robinlovelace** [comment]
I think it is a regression. Can provide details..

**Comment by ronblum** [comment]
This is similar to behavior in earlier releases. I tried with RStudio Desktop 1.4.1717, and the same happens. The difference is that in 2022.06.0-daily+530, "Run All" stops after the first failure (chunk 2), while 1.4.1717 tries to run them all:

```
> # Chunk 2
> name = "Jackie"
> print(f"hello {name}")
Error: unexpected string constant in "print(f"hello {name}""
> quit
function (save = "default", status = 0, runLast = TRUE) 
.Internal(quit(save, status, runLast))
<bytecode: 0x7fa2bb63c478>
<environment: namespace:base>
> 
> # Chunk 3
> pwd
Error: object 'pwd' not found
```

Note: I added `quit` in chunk 2 so that the processing doesn't stop with Python using `reticulate::repl_python`.

**Comment by Robinlovelace** [comment]
Confirmed. This bug is present in 

RStudio 2022.06.0-daily+404 "Spotted Wakerobin" Daily (70299b91d6afd46cfaece3d245c3bd3e702a1d75, 2022-05-13) for Ubuntu Bionic
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) QtWebEngine/5.12.8 Chrome/69.0.3497.128 Safari/537.36

**Comment by Robinlovelace** [comment]
And, to my surprise, previous versions of RStudio.

**Comment by Robinlovelace** [comment]
Nudge on this, it's a substantial bug. Hopeful for a fix before 2023-02-01, as per milestone, but great to see a timeline for this.

**Comment by JacobBumgarner** [comment]
FYI, the `rstudio-server-2024.04.2-764-amd64` RStudio update fixed this problem for me.
